### PR TITLE
Add Basic UI

### PR DESCRIPTION
**Pull request by umair-rao**
### General requirements
✔️ Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
✔️ Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✔️ Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Ruby requirements
✔️ Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
### Project requirements
✔️ Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
### UI example
✔️ Your console app should behave in the same way as it is in the example.
✔️ Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
✔️ List all books.
✔️ List all people.
✔️ Create a person (teacher or student, not a plain Person).
✔️ Create a book.
✔️ Create a rental.
✔️ List all rentals for a given person id.
✔️ In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
✔️ Present the user with a list of options to perform.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    puts 'Rental created successully'
+  end
+
+  def list_rentals
+    puts 'ID of person:'
+    pr_id = gets.chomp.to_i
+    @rentals.each do |rent|
+      if rent.person.id == pr_id
+        puts "Date:#{rent.date}, Book #{rent.book.title} by #{rent.book.author} borrowed by #{rent.person.name}"
```
Comment (Munsa1) [inline app.rb]: - [ ] When the user tries to list rentals by ID, the person in the rental list is currently not being displayed. It is showing `Unkow`. Kindly fix this.
image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      puts ' '
+    end
+  end
+
+  def list_all_people(_people)
+    @people.each do |person|
+      print "[#{person.class}] name: #{person.name}, ID: #{person.id}, #{person.age}"
+      puts ' '
+    end
+  end
```
Comment (Munsa1) [inline app.rb]: - [ ] When listing people, the user is getting `Unkow` fro the student, kindly fix this

image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @umair-rao, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Implemented UI as In given video example :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Kindly caplitalize the first letter of every word in your commit messages

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    case permit
+    when 'Y', 'y'
+      permission = true
+    when 'N', 'n'
+      permission = false
+    end
+    student = Student.new(age, name, parent_permission: permission)
+    @people.push(student)
+    puts 'Person successfully created'
+  end
```
Comment (omar-muhamad) [inline app.rb:49]: - [ ] Same problem still exists, When I create a new student and list all students I get unknown for the name like the image below, Please fix that

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @umair-rao,
Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @umair-rao , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] As requested by the previous reviewer, i am still getting that error:-

Screenshot from 2023-02-19 00-44-49
- Note:- In the `student.rb` file there is a mismatch in the order of parameter passing this should be according to the parameter you are passing in the main.rb file while creation.
- Kindly use 
```
def initialize(age, name = 'Unknow', parent_permission = true,classroom)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
```
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @umair-rao 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested was implemented ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**